lib/scholar/metrics/distance.ex

```diff
@@ -35,16 +40,31 @@ defmodule Scholar.Metrics.Distance do
       iex> y = Nx.tensor([1, 2, 3])
       iex> Scholar.Metrics.Distance.euclidean(x, y)
       ** (ArgumentError) expected input shapes to be equal, got {2} != {3}
+
+      iex> x = Nx.tensor([[1,2], [3,4]])
+      iex> y = Nx.tensor([[8,3], [2,5]])
+      iex> Scholar.Metrics.Distance.euclidean(x, y, axes: [0])
+      #Nx.Tensor<
```

Comment:
Could you use a {3,2}-shaped example to make it more clear what `axes: [0]` means?

Also, a multi-axis example could be useful as well.

Reply:
This is also relevant to the other functions as well